include/xtensor/xexpression.hpp

```diff
@@ -156,29 +156,26 @@ namespace xt
      * evaluation_strategy *
      ***********************/
 
+    namespace detail
+    {
+        struct option_base {};
+    }
+
     namespace evaluation_strategy
     {
-        struct base
-        {
-        };
 
-        struct immediate : base
-        {
-        };
-        
-        struct lazy : base
-        {
-        };
-        
+        struct _immediate : xt::detail::option_base {};
```

Comment:
names starting with `_` are reserved for the standard. This could be replaced with `immediate_t` for instance.